include/lama/ros/loc2d_ros.h

```diff
@@ -141,6 +143,9 @@ class Loc2DROS {
     Pose2D odom_;
     Loc2D::Options options_;
     Pose2D initial_prior_;
+    Pose2D current_pose_;
+    tf::Quaternion current_orientation_;
+    geometry_msgs::PoseWithCovarianceStamped cur_pose_msg_;
```

Comment:
We do not need to keep track of these 3 variables. The Pose of the robot should be published only after its calculation.

This is a semantic issue. The `tf` will have a "continuous" version of the robot's pose, while the topic `~pose` should contain only the pose calculated by the Localization process.

Reply:
- **`geometry_msgs::PoseWithCovarianceStamped cur_pose_msg_;`**
  In general I like to keep memory allocated, so I allocate everything in constructors. With the correct compiler flags being set, even we allocate memory for an object in a function, it is allocated and reused after those function calls. But this time I did it for another reason: to avoid setting; 
  ```cpp
  cur_pose_msg_.header.frame_id = global_frame_id_;
  ```
  everytime we publish this message, I made it a member since this will never change.

- **`Pose2D current_pose_;`**
  You are right about having it since getPose() returns a reference, sorry my bad. I removed it.

- **`tf::Quaternion current_orientation_;`***
  But about this quaternion I'd like to keep it and avoid calling `tf::createQuaternionFromYaw(double yaw)` and calculating the quaternion twice in every update:
  - first in `onLaserScan()`
  - second in `publishCurrentPose()`
  
- **The pose topic logic**
  > This is a semantic issue. The tf will have a "continuous" version of the robot's pose, while the topic ~pose should contain only the pose calculated by the Localization process.

  I somehow agree with this semantic issue but I have a robot that travels between floors and changes maps. `amcl` follows the the same semantic. Which is quite ok for me since my robot's software stack requires the current belief information of the robot being in the `pose` topic. Even though robot didn't move and no updates have been computed, I still need the initial pose values in the `pose` topic. This way I can start navigating and knowing my pose in the beginning of the time that a new instances of move_base and map_server is started running. This way my software stack works with `amcl` and `loc_2d_ros` just fine.
  
  `tf` requires informations to have an updated timestamp, so even though no update, we still publish information to it. But this `pose` topic is for just informing other nodes about last known pose and doesn't require the updated timestamp information. `amcl` stops publishing to this topic when there is no update, also publishes once after every manual pose estimate message.

Reply:
> Even though robot didn't move and no updates have been computed, I still need the initial pose values in the pose topic

Fair enough